api/serializers.py

```diff
@@ -35,8 +39,46 @@ class Meta:
         model = User
         fields = ("token", "username", "password")
 
-
 class VideoSerializer(serializers.ModelSerializer):
+    def create(self, validated_data):
+        data = self.context.get("data")
+        sender = self.context.get("user")
+
+        # receiver = User.objects.get(id=validated_data.pop('receiver'))
+        # Data is prepended by "data:video/mp4;base64,", need to remove those 22 characters
+        # Encode converts from string to bytes
+        data_bytes = data[22:].encode("ascii")
+        instance = self.Meta.model(**validated_data)
+
+        instance.sender = sender
+
+        # We need to do this before save because we use it for fname
+        instance.created_at = now()

```

Comment:
Do you mean with `auto_now_add=True`? That was not working for me because that is applied when `save()` is called.

Reply:
Yeah, so that might change if we go a different naming strategy. TBH the name doesn't matter a ton, although I can see the value of metadata at some point